src/main/java/hudson/tasks/junit/TestObject.java

```diff
@@ -67,12 +67,14 @@
    
     public abstract TestObject getParent();
 
-	public abstract String getId(); 	
-	/**
-	 * Returns url relative to TestResult.
+	public abstract String getId();
+
+    /**
+     * Returns the URL of this {@link TestObject}, relative to the context root.
      *
-     * @return the url relative to {@link TestResult}.
```

Comment:
Perhaps we could deprecate the old method and introduce a replacement intended to have different behavior?

Reply:
at first glance I think this will break at least the https://github/com/jenkinsci/cucumber-testresult-plugin

Reply:
I had documented the problem from
https://issues.jenkins-ci.org/browse/JENKINS-36305?focusedCommentId=262717&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-262717
I don't see how it can be fixed otherwise.